zcash_proofs/src/lib.rs

```diff
@@ -118,11 +120,27 @@ pub fn load_parameters(
     let sprout_fs =
         sprout_path.map(|p| File::open(p).expect("couldn't load Sprout groth16 parameters file"));
 
-    let mut spend_fs = hashreader::HashReader::new(BufReader::with_capacity(1024 * 1024, spend_fs));
-    let mut output_fs =
-        hashreader::HashReader::new(BufReader::with_capacity(1024 * 1024, output_fs));
-    let mut sprout_fs =
-        sprout_fs.map(|fs| hashreader::HashReader::new(BufReader::with_capacity(1024 * 1024, fs)));
+    parse_parameters(
+        BufReader::with_capacity(1024 * 1024, spend_fs),
+        BufReader::with_capacity(1024 * 1024, output_fs),
+        sprout_fs.map(|fs| BufReader::with_capacity(1024 * 1024, fs)),
+    )
+}
+
+fn parse_parameters<R: io::Read>(
+    spend_fs: R,
+    output_fs: R,
+    sprout_fs: Option<R>,
+) -> (
```

Comment:
nit: I know this isn't newly introduced, but returning a tuple here seems unwise, since there are two `Parameters<Bls12>` returned and up to three `PreparedVerifyingKey<Bls12>` returned, and it seems like it'd be really easy to mix up which one was for what. Maybe either use newtype wrappers to distinguish the spend vs output params, or use a record type, or both?

Reply:
Opened #252 for this.